llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -3944,6 +3944,30 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     }
   }
 
+  /// Handle intrinsics by applying the intrinsic to the shadows.
+  /// The origin is approximated using setOriginForNaryOp.
+  ///
+  /// For example, this can be applied to the Arm NEON vector table intrinsics
+  /// (tbl{1,2,3,4}).
+  void handleIntrinsicByApplyingToShadow(IntrinsicInst &I, unsigned int numArgOperands) {
+    IRBuilder<> IRB(&I);
+
+    // Don't use getNumOperands() because it includes the callee
+    assert (numArgOperands == I.arg_size());
```

Comment:
why not to use arg_size() ?
and simplify cases:

```
    case Intrinsic::aarch64_neon_tbl1: {
    case Intrinsic::aarch64_neon_tbl2: {
    case Intrinsic::aarch64_neon_tbl3: {
    case Intrinsic::aarch64_neon_tbl4: {
      handleIntrinsicByApplyingToShadow(I);
      break;
    }
```

Reply:
It was there as a correctness check (we know that neon_tbl1 should have 2 arguments, neon_tbx1 should have 3 arguments). I will remove it.